Oauth social media Login

pavonis9 opened the pull request:
Social media Oauth login fake popup. a JS popup will be launched only in Mac OS and Windows. For the other OSs (Linux, Android, IOS ) it will be redirected to a Facebook login page (twitter login and google+ will be available in the next update).

Original idea from : 
https://github.com/wifiphisher/wifiphisher/issues/1075

pavonis9 committed:
Social media Login

Social media Oauth login fake popup (Mac OS, Windows,Linux, Android, IOS Friendly), Notice that the popup will be launched only in Mac OS and Windows. For the other OSs it will be redirected to a facebook login page (twitter login and google+ will be available in the next update).

sophron commented:
Thanks for contributing. I'll have a look soon.

sophron commented:
I had a look. I like how you imitate the browser and the HTTPS bar but I think the index page is not very realistic. Here's what I suggest:

Instead of creating a new scenario, we can simply improve our existing "oauth-login" scenario. If the victim client is on MacOS or Windows, then we will display the "Connect to Facebook" button as in your scenario. If the victim user is on any other OS (e.g. Android), then we leave the page as it is with the existing HTML form.

Feel free to update your PR with the changes and I will do a second review.

pavonis9 commented:
For the index I know it's not that realistic because I started the project from scratch if we found some good resources that will be useful.
And for the changes you mentioned I'll do my best and try to make it happen.

pavonis9 committed:
Merge pull request #1 from wifiphisher/master

update

pavonis9 committed:
Enhanced Oauth login

Added Facebook popup login and the option to enable or disable it from config.ini

pavonis9 commented:
I

sophron commented:
I had a look at the recent changes and I think it's much better now. Good job.

I'll have a closer look soon and let you know what needs to be changed.

sophron reviewed (changes requested):
Hi,

Thank you for working on this. 

We will have to improve the code quality before merging this PR. Imagine other contributors that will want to extend your code after some time. They should find your code structured and self-explanatory. Some things to consider:

* Use proper names for variables and functions that are descriptive.
* Don't repeat the same code again and again. 
* Make sure that the code is structured properly. For example, you can have a directory for each browser that you want to emulate (e.g. "firefox", "safari", etc) and put all the relevant code in there. There should be one main `index.html` file that should act as a "controller" that imports the right files accordingly
* Write as many comments as possible.

When you update your code, I'll perform a second round of review.

pavonis9 commented on wifiphisher/data/phishing-pages/socialmedia-beta/html/static/css/fonts.css:
```diff
@@ -0,0 +1,21 @@
+@font-face {
```
I'll remove the unused CSS lines.

pavonis9 commented on wifiphisher/data/phishing-pages/socialmedia-beta/config.ini:
```diff
@@ -0,0 +1,6 @@
+[info]
+Name: Social media Login (beta)
+Description: Facebook oauth login fake popup (Mac OS, Windows,Linux, Android, IOS Friendly), Notice that the popup will be launched only in Mac OS and Windows. For the other OSs it will be redirected to a facebook login page (twitter login and google+ will be available in the next update).
```
This is the old template I'll remove it and keep only the new Oauth login scenario.

sophron replied:
Yes.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/html/index.html:
```diff
@@ -1,25 +1,70 @@
+window.location='login.html';
+    };
+if (OSName == "MacOS") {
+window.location='signin.html';
```
I'll do the Mac-OS and Windows login only in one page.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/html/index.html:
```diff
@@ -1,25 +1,70 @@
+if (OSName == "UNIX") {
+    return false;
+    };
+}
```
Yeah I'm adding now comments in the update.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/html/index.html:
```diff
@@ -1,25 +1,70 @@
+
+
+if (OSName == "Windows") {
+window.location='login.html';
```
I'll merge the Mac-OS and Windows login in one page.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/config.ini:
```diff
@@ -1,6 +1,10 @@
 organization:
+popup_value: true
+# If you want to use the popup login form (Compatible only with Windows and Mac-OS) set the popup_value to true.
+# If you want to disable the popup set the popup_value to false.
```
I'm working on it everything will be ready soon.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/html/static/css/style2.css:
```diff
@@ -0,0 +1,217 @@
+body {
```
Why there is two style.css : The first one is for the popup page (login.html) and the second one is for the default oauth scenario if I used only one sytle.css it will be some conflicts in both pages. I'll rename the second one.

sophron replied:
Yes, if you want to have two, we need a more descriptive name (e.g. `style-popup.css` and `style-form.css`)

pavonis9 replied:
That's a good idea. I'll do it on the next update.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/html/signin.html:
```diff
@@ -0,0 +1,296 @@
+
+<script>
+function alertee() {
+  alert("Twitter is not available for the moment!");
```
I have a better idea. What about a variable to choose either from this alert or do nothing like you said.

sophron replied:
Sure.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/html/nointernet.html:
```diff
@@ -0,0 +1,2 @@
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<p>User Didn't Sign In</p>
```
Once the user press the cancel button on the js popup he will be redirected to a new page and this message will be showed.
Of course this one is used only to force the user to connect because once he sees this message then he'll think that the provided service is real and he'll know that there is no other way to connect besides from this.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/html/signin.html:
```diff
@@ -0,0 +1,296 @@
+        frames[0].requestFocus();
+    }
+
+  function openthegoddamnwindow() {
```
Definitely will do.

pavonis9 commented on wifiphisher/data/phishing-pages/oauth-login/config.ini:
```diff
@@ -1,6 +1,10 @@
-Name: OAuth Login Page
-Description: A free Wi-Fi Service asking for Facebook credentials to authenticate using OAuth
+Name: Enhanced OAuth Login Page
+Description: A free Wi-Fi Service asking for Facebook credentials using default facebook login page to authenticate using OAuth
```
Okey I'll make the changes.

pavonis9 committed:
Delete config.ini

pavonis9 committed:
Delete text.html

pavonis9 committed:
Delete oauth.html

pavonis9 committed:
Delete nointernet.html

pavonis9 committed:
Delete login.html

pavonis9 committed:
Delete index.html

pavonis9 committed:
Delete facebook.html

pavonis9 committed:
Delete ssl.png

pavonis9 committed:
Delete jsframe.js

pavonis9 committed:
Delete fb.jpg

pavonis9 committed:
Delete jquery.min.js

pavonis9 committed:
Delete font-awesome.css

pavonis9 committed:
Delete font-awesome.min.css

pavonis9 committed:
Delete normalize.min.css

pavonis9 committed:
Delete fonts.css

pavonis9 committed:
Delete FontAwesome.otf

pavonis9 committed:
Delete Lato-700.woff2

pavonis9 committed:
Delete Lato-regular.woff2

pavonis9 committed:
Delete fontawesome-webfont.eot

pavonis9 committed:
Delete fontawesome-webfont.svg

pavonis9 committed:
Delete fontawesome-webfont.ttf

pavonis9 committed:
Delete fontawesome-webfont.woff

pavonis9 committed:
Delete fontawesome-webfont.woff2

pavonis9 committed:
Delete fm5arwc28y.png

pavonis9 committed:
Update OAuth login phishing scenario

In this update, wifiphisher user has the ability to customize the popup as he likes in config.ini

pavonis9 committed:
Delete signin.html

pavonis9 committed:
Delete nointernet.html

pavonis9 committed:
Delete fb.jpg

pavonis9 committed:
Delete style2.css

pavonis9 commented:
@sophron I just cleaned up the unused files of the old commits and all changes are ready now.

I commented every details you need to know they are mentioned in config.ini and inside the html files.

sophron commented:
Okay, I will review again soon.

sophron commented:
Hi,

The code looks much better than before. A few questions:

* From what I understand, `style.css` and `stylejs.css` are both custom CSS files (i.e. both were written by Wifiphisher developers). Does it make sense to merge them?

* You are using a number of fonts. Do we need all of them? Note that we don't want to have the exact same page as Facebook (because of copyright issues), but something similar to it.

* There are a number of fonts under `html/static/fonts/` and then I see more fonts under `html/css`. Then, there is a `jsframe.js` that is NOT under the `javascript` directory.

pavonis9 commented:
>     * From what I understand, `style.css` and `stylejs.css` are both custom CSS files (i.e. both were written by Wifiphisher developers). Does it make sense to merge them?
> 
Yeah I have two Ideas and only one of them which gonna work: So the first one is trying to rename the parts who have conflicts and then merge the style.css files or I'll create one principal style.css which cover both pages then I'll put the parts who have conflicts in the <script></script> part of each page.
>     * You are using a number of fonts. Do we need all of them? Note that we don't want to have the exact same page as Facebook (because of copyright issues), but something similar to it.
> 
Okay, I'll take a look and make the changes.
>     * There are a number of fonts under `html/static/fonts/` and then I see more fonts under `html/css`. Then, there is a `jsframe.js` that is NOT under the `javascript` directory.
>
My bad I'll move jsframe.js into `javascript` directory.

pavonis9 committed:
Delete 2eer.otf

pavonis9 committed:
Delete FontAwesome.otf

pavonis9 committed:
Delete Lato-700.woff2

pavonis9 committed:
Delete Lato-regular.woff2

pavonis9 committed:
Delete fontawesome-webfont.eot

pavonis9 committed:
Delete fontawesome-webfont.svg

pavonis9 committed:
Delete fontawesome-webfont.ttf

pavonis9 committed:
Delete fontawesome-webfont.woff

pavonis9 committed:
Delete fontawesome-webfont.woff2

pavonis9 committed:
Delete jsframe.js

pavonis9 committed:
Delete ufonts.com_klavika-bold-opentype-webfont.woff

pavonis9 committed:
Delete ufonts.com_klavika-bold-opentype-webfont.woff2

pavonis9 committed:
Delete stylejs.css

pavonis9 committed:
New changes

Merging CSSs files into one style.css and move the parts having conflicts inside the <style><style> of the pages.
Removing the unused fonts folder.
Adding a template for the users who want to use their own page with the popup.
Creating a full detailed documentation for the people who want to use their own page inside the popup (The Documentation is inside template.html).
jsframe.js has been moved to the js folder.

sophron commented on wifiphisher/data/phishing-pages/oauth-login/html/template.html:
```diff
@@ -0,0 +1,279 @@
+The source of the used JS library : https://github.com/riversun/JSFrame.js
+For further information about using the JS popup check : https://github.com/riversun/JSFrame.js
+*/
+
```
This documentation is nice but the best place for it is on `config.ini` file.

sophron commented on wifiphisher/data/phishing-pages/oauth-login/html/static/css/fonts.css:
```diff
@@ -0,0 +1,21 @@
+  src: url('/fonts/Lato-regular/Lato-regular.eot?#iefix') format('embedded-opentype'),
+       local('Lato Regular'),
+       local('Lato-regular'),
+       url('../fonts/Lato-regular.woff2') format('woff2');
```
The "fonts" directory no longer exists.

sophron commented on wifiphisher/data/phishing-pages/oauth-login/html/static/css/fonts.css:
```diff
@@ -0,0 +1,21 @@
+  src: url('/fonts/Lato-700/Lato-700.eot?#iefix') format('embedded-opentype'),
+       local('Lato Bold'),
+       local('Lato-700'),
+       url('../fonts/Lato-700.woff2') format('woff2');
```
The "fonts" directory no longer exists.

sophron commented on wifiphisher/data/phishing-pages/oauth-login/html/static/css/fonts.css:
```diff
@@ -0,0 +1,21 @@
+  font-weight: 400;
+  font-style: normal;
+  src: url('/fonts/Lato-regular/Lato-regular.eot');
+  src: url('/fonts/Lato-regular/Lato-regular.eot?#iefix') format('embedded-opentype'),
```
The "fonts" directory no longer exists.

sophron commented on wifiphisher/data/phishing-pages/oauth-login/html/static/css/style.css:
```diff
@@ -1,23 +1,70 @@
+  width:980px;
+  padding-top:20px;
+}
+.login2-btn {
```
There is no `login2-btn`. We can remove this.

sophron commented:
This PR is in a much better state now. If you can remove unnecessary code and add the HTML code of the popup box on the `config.ini`, we will be ready to merge it.

sophron commented:
I'll continue the work on this and merge it soon.
